### ISSUE TITLE
Bug: Cross-encoder raw logits not normalized in search_components/reranking.py

### ISSUE THREAD

**Issue by mrveiss** [mrveiss/AutoBot-AI#1533]
## Problem

`knowledge/search_components/reranking.py:_apply_rerank_scores()` (line 43) stores raw cross-encoder logits directly as `rerank_score` without sigmoid normalization:

```python
result["rerank_score"] = float(scores[i])  # Raw logits — can be negative or >1
```

Additionally, line 47 **overwrites** the original ChromaDB similarity score:
```python
result["score"] = result.get("rerank_score", 0)  # Replaces real similarity with raw logit
```

## Impact

- MS MARCO `ms-marco-MiniLM-L-6-v2` returns unbounded logits (e.g., -2.5 to +3.0)
- After reranking, `result["score"]` contains raw logits instead of 0-1 normalized scores
- Any downstream code filtering by `min_score` or `score >= threshold` gets broken values
- Used by `SearchMixin._rerank_results()` → `enhanced_search` and `enhanced_search_v2` API endpoints

## Contrast with #1526 Fix

The same cross-encoder model in `advanced_rag_optimizer.py` was fixed in #1526 with sigmoid:
```python
normalized_ce = 1.0 / (1.0 + math.exp(-float(ce_score)))
```

This fix was NOT applied to the parallel reranking path in `search_components/reranking.py`.

## Fix

1. Apply sigmoid to logits before storing as `rerank_score`
2. Combine normalized score with original similarity instead of overwriting

## Discovered During
Implementing fix for #1526
